first_order_model/fom_api_test.py

```diff
@@ -5,127 +5,266 @@
+                    action='store_true',
+                    help="encode high-res video stream with vpx")
+parser.add_argument("--lr-quantizer",
+                    type=int, default=32,
+                    help="quantizer to compress low-res video stream with")
+parser.add_argument("--encode-lr",
+                    action='store_true',
+                    help="encode low-res video stream with vpx")
 parser.set_defaults(verbose=False)
-opt = parser.parse_args()
-
-def visual_metrics(driving, prediction):
-    start = torch.cuda.Event(enable_timing=timing_enabled)
-    start.record()
-    driving = np.expand_dims(img_as_float32(driving), 0)
-    prediction = np.expand_dims(img_as_float32(prediction), 0)
-    
-    driving = np.transpose(driving, [0, 3, 1, 2])
-    prediction = np.transpose(prediction, [0, 3, 1, 2])
-    
-    original_tensor = torch.from_numpy(driving)
-    prediction_tensor = torch.from_numpy(prediction)
-    if torch.cuda.is_available():
-        original_tensor = original_tensor.cuda()
-        prediction_tensor = prediction_tensor.cuda()
-
-    lpips_val = vgg_model.compute_loss(original_tensor, prediction_tensor).data.cpu().numpy().flatten()[0]
-    ssim = piq.ssim(original_tensor, prediction_tensor, data_range=1.).data.cpu().numpy().flatten()[0]
-    psnr = piq.psnr(original_tensor, prediction_tensor, data_range=1., \
-            reduction='none').data.cpu().numpy().flatten()[0]
-
-    end = torch.cuda.Event(enable_timing=timing_enabled)
-    end.record()
-    torch.cuda.synchronize()
-    curr_time = start.elapsed_time(end)
-    return lpips_val, ssim, psnr
-
-def get_lr_array(input_array, lr_size, device):
-    lr_array = frame_to_tensor(img_as_float32(input_array), device)
-    lr_array = F.interpolate(lr_array, lr_size).data.cpu().numpy()
-    lr_array = np.transpose(lr_array, [0, 2, 3, 1])[0]
-    lr_array *= 255
-    lr_array = lr_array.astype(np.uint8)
-    return lr_array
-
-model = FirstOrderModel(opt.config)
-use_lr_video, lr_size = model.get_lr_video_info()
-video_name = opt.video_path
-source = np.random.rand(model.get_shape()[0], model.get_shape()[1], model.get_shape()[2])
-source_kp, _= model.extract_keypoints(source)
-model.update_source(0, source, source_kp)
+args = parser.parse_args()
+
+main_configs = get_main_config_params(args.config)
+generator_type = main_configs['generator_type']
+use_lr_video = main_configs['use_lr_video']
+lr_size = main_configs['lr_size']
+print(main_configs)
+
+video_duration = get_video_duration(args.video_path)
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
-if use_lr_video:
-    source_lr = get_lr_array(source, lr_size, device)
 
-predictions = []
-times = []
-lpips_list = []
-psnr_list = []
-ssim_list = []
+if not os.path.exists(args.log_dir):
+    os.makedirs(args.log_dir)
+
+# model initialization and warm-up
+if generator_type not in ['vpx', 'bicubic']:
+    model = FirstOrderModel(args.config, args.checkpoint)
```

Comment:
blanking on where the definition for this is. Can you point me to it?

Reply:
`generator_type` is in the configs, vpx and bicubic also have configs

Reply:
default configs don't have it and it means `occlusion_aware`, generator_type is used in aiortc to deal with the received frame